server/src/core/stat/stat-service.ts

```diff
@@ -0,0 +1,82 @@
+import { Db, Collection } from 'mongodb';
+import { createDbOpFailedError, rethrowIfAppError, createResourceNotFoundError } from '../error';
+import { generateId } from '../util';
+import * as messages from '../messages';
+import { UserService }  from '../user';
+import { TransactionService } from '../payment';
+import { StatService, DbStat, StatsArgs, Stat, StatCreateArgs, StatUpdateArgs } from './types';
+
+const COLLECTION = 'stats';
+export class Stats implements StatService {
+  private db: Db;
+  private collection: Collection<DbStat>;
+  private users: UserService;
+  private transactions: TransactionService;
+
+  constructor(db: Db, args: StatsArgs) {
+    this.db = db;
+    this.collection = this.db.collection(COLLECTION);
+    this.users = args.users;
+    this.transactions = args.transactions;
+    // Compute and/or update stats every 5 minutes 
+    let intervalId = setInterval(this.compute, 5 * 60 * 1000);
```

Comment:
The computation job shouldn't be handled here. This service should expose a method to retrieve the stats (`get()`) and a method to update it (`update()`). The `get()` method will be called by the REST endpoint, but the `update()` should be called by the worker jobs. The worker is where the interval should be applied. Check how we're running the distribution job for reference.